src/CBT.Core/Internal/DefaultNuGetDownloader.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Microsoft.Build.Framework;
+using System;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CBT.Core.Internal
+{
+    /// <summary>
+    /// A class that will download the latest version of NuGet from NuGet.org.
+    /// </summary>
+    internal static class DefaultNuGetDownloader
+    {
+        private static readonly Uri NuGetUrl = new Uri("https://dist.nuget.org/win-x86-commandline/latest/nuget.exe");
```

Comment:
I wonder if we want to consider passing a list of uri's to down load.  Basically to untie this all from nuget and it becomes a basic downloader given a set of uri's